docs/devel/offline-virtual-machine.md

```diff
@@ -0,0 +1,234 @@
+metadata:
+  name: myvm
+spec:
+  running: false
+  template:
+    metadata:
+      labels:
+        my: label
+    spec:
+      domain:
+        resources:
+          requests:
+            memory: 8Mi
+        devices:
+          disks:
+          - name: disk0
+            volumeName: mypcv
+      volumes:
+        - name: mypvc
+          persistentVolumeClaim:
+            claimName: myclaim
+```
+
+The file specification follows the Kubernetes guide. The apiVersion is linked
+with the Kubevirt release cycle.
+
+In the metadata section, there is a *required* field, the **name**. Then
+following the spec section, there are two important parts. The **running**, which
+indicates the current state of the VirtualMachine attached to this VirtualMachine.
+Second is the **template**, which is the VirtualMachine template.
+
+Let us go over each of these fields.
+
+### Name
+
+The name if important for the OfflineVirtualMachine, it is used to find created
+VirtualMachine. The VirtualMachine name is directly derived from
+the OfflineVirtualMachine. This means, if OfflineVirtualMachine is names 'testvm',
+the VirtualMachine is named 'testvm' too.
+
+When the VirtualMachineStatefulSet is created from the OfflineVirtualMachine,
+its name is also derived. In the case of 'testvm', name of each virtual machine
+in the stateful set is prefixed with the 'testvm' and appended with the ordering:
+'testvm0', 'testvm1' and so on.
+
+### Template
+
+The template is directly the VirtualMachine object. It contains the metadata
+section and a full VirtualMachineSpec.
+Thus it is identical to a VirtualMachine without a VirtualMachineStatus.
+This follows the same pattern which Kubernetes has on such template sections.
+There a PodTemplate contains of Metadata and a PodSpec.
+
+## OfflineVirtualMachine behaviour
+
+The OfflineVirtualMachine has to be in sync with its VirtualMachine. This means
+that the OfflineVirtualMachine has to observe both, the OfflineVirtualMachine
+and created VirtualMachine. When the link is established the config changes
+are translated to the VirtualMachine, and coresponding status changes are
+translated back to OfflineVirtualMachine.
+
+### Status
+
+The status of the VirtualMachine is watched and is reflected in OfflineVirtualMachine status.
+The status contains the current state of the VirtualMachine (running/not running)
+and whether it is ready.
+
+More detailed information is queried directly from the status of the VirtualMachine.
+
+### OwnerReference
+
+Linking the created VirtualMachine to its parent OfflineVirtualMachine pose
+a challenge. Using the same name is only part of the solution. To find the
+parent OfflineVirtualMachine programatically in the Kubernetes, the OwnerReference
+is used. As described in the [design document](https://github.com/kubernetes/community/blob/930ce65595a3f7ce1c49acfac711fee3a25f5670/contributors/design-proposals/api-machinery/controller-ref.md), the OwnerReference lives in the
```

Comment:
Please use a link without a commithash.
The commithash link will alw<ys work, but we won't know if this bnecomes outdated.